### PR TITLE
Add -walletupdates: resolves #12632

### LINKED ISSUE #12632: Add a way to skip rescanning on startup
I sometimes need to attach different wallets to my Bitcoin Core instance only to use certain RPC calls. I do not need up to date transaction or balance information. Basically I only want to perform `dumpwallet`. Would it be feasible to add a startup mode where rescanning and other optional work is skipped? Maybe block verification could also be skipped.

Rescanning takes about 2h on my fairly recent CPU on an SSD. It would be very useful if this step could be skipped.

### PR DESCRIPTION
**Pull request by skeees**
When false, wallet does not rescan the chain from its last known best block on startup or subscribe to CValidationInterface notifications.  Useful for wallet maintenance type activities where one desires to avoid the overhead of rescan on startup.

Fixes #12632.

### CODE REVIEW

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -4034,17 +4041,26 @@ CWallet* CWallet::CreateWalletFromFile(const std::string& name, const fs::path&
...
     {
-        CWalletDB walletdb(*walletInstance->dbw);
-        CBlockLocator locator;
-        if (walletdb.ReadBestBlock(locator))
-            pindexRescan = FindForkInGlobalIndex(chainActive, locator);
+        if(subscribe_to_chain_updates) {
+            CWalletDB walletdb(*walletInstance->dbw);
+            CBlockLocator locator;
+            if (walletdb.ReadBestBlock(locator))
+                pindexRescan = FindForkInGlobalIndex(chainActive, locator);
```
Comment (MarcoFalke) [inline src/wallet/wallet.cpp]: Might want to add braces `{}` around that to adhere to the coding style for new code and then install `clang-format` and run the https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools#clang-format-diffpy script to preempt whitespace nitpicking.

**Comment by laanwj** [comment]
Needs a test in `test/functional`? (possibly as part of another wallet test)

**Comment by skeees** [comment]
Will add functional test and update

**Comment by skeees** [comment]
Test added - can rebase and squash into one commit if there are no further comments / fixes

**Comment by skeees** [comment]
@jnewbery @promag - with the ongoing efforts to do dynamic wallet (un)loading and to move wallet maintenance into a wallet-tool - is this still relevant/necessary? if not will close

**Comment by promag** [comment]
Not sure @skeees, this is orthogonal to dynamic wallet (un)loading.

BTW, how about a different approach: load the wallet in "read only" mode?

**Comment by luke-jr** [comment]
Option name seems too much like `-upgradewallet`

Maybe `-walletsync` instead?

**Pull request closed without merging**